src/com/puppetlabs/puppetdb/facts.clj

```diff
@@ -11,17 +11,37 @@
 
 ;; SCHEMA
 
+;; Terminology:
+;;   facts: {"kernel" "Linux" ...}
+;;   factmap: {:factset_id ... :fact_path_id ... fact_value_id ...}
+;;   fact-path path: ["foo" "bar"]
+;;   pathstr: "foo~#bar..."
+;;   pathmap: {... :path ... :name ... :depth ...}
+;;   valuemap: {... :value_type_id ... :value_boolean ...}
```

Comment:
Right, this is one of the bits I figured I might delete from the final version -- I just added it while I was working so I could keep track.  And I left it in for now to possibly spark this discussion.

Another couple of things I mentioned to @senior are that it might be nice to have somewhat more consistent naming of types in the code, and that it might be nice to have some approach to naming schemas in particular, wether via Capitalization (which he mentioned some people do), or a naming convention like valuemap-scm, or putting them all in separate namespaces.  Having (for example) both factpath and fact-path scattered around seems a bit muddled, and I think I might prefer to save the really obvious names like fact, path, or value, for local-var/global-var/function names, rather than schemas.